pkg/api/validation/validation.go

```diff
@@ -116,6 +120,22 @@ func ValidateAnnotations(annotations map[string]string, fldPath *field.Path) fie
 	if annotations[api.AffinityAnnotationKey] != "" {
 		allErrs = append(allErrs, ValidateAffinityInPodAnnotations(annotations, fldPath)...)
 	}
+
+	hostname, exists := annotations[utilpod.PodHostnameAnnotation]
```

Comment:
Please put these in the Pod and Endpoints validation functions.  The Affinity one is wrong.